### PR TITLE
[Optimize] Add session variable `max_fetch_remote_schema_tablet_count` to limit tablets size for remote schema fetch

### PR DESCRIPTION
**Pull request by eldenmoon**
Describing tables with many partitions and tablets can cause high CPU usage. To mitigate this, we estimate and pick sample tablets for schema fetch, reducing the overall cost.

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 1]
```diff
@@ -2761,6 +2762,50 @@ public List<Tablet> getAllTablets() throws AnalysisException {
...
+                // compare with desc order
+                return Long.compare(p2.getId(), p1.getId());
+            }
+        }).collect(Collectors.toList());
+
+        // 3. Collect tablets from partitions
+        for (Partition partition : sortedPartitions) {
+            List<Tablet> targetTablets = new ArrayList<>(partition.getBaseIndex().getTablets());
+            Collections.shuffle(targetTablets);
+
```
Comment (csun5285) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:2791]: `if (!targetTablets.isEmpty() && targetTablets[0].getReplicas().getVersion() > 1)`, 
some partitions might have no data

**Review by csun5285 — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java (hunk 1 above)
Comment (eldenmoon) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java]: nice catch.done

**Review by csun5285 — APPROVED** [review]
LGTM

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**